src/ProductConstructionService/ProductConstructionService.FeedCleaner/FeedCleanerJob.cs

```diff
@@ -45,61 +44,104 @@ public async Task CleanManagedFeedsAsync()
         {
             _logger.LogInformation("Processing feeds for {account}...", azdoAccount);
 
-            List<AzureDevOpsFeed> allFeeds;
-            try
+            var (packageFeeds, symbolFeeds) = await FetchFeeds(azdoAccount);
+
+            _logger.LogInformation("Processing {feedCount} package feeds...",
+                packageFeeds.Count);
+
+            int feedsCleaned = await ProcessFeedsInParallelAsync(packageFeeds, async (scope, feed) =>
             {
-                allFeeds = await _azureDevOpsClient.GetFeedsAsync(azdoAccount);
-            }
-            catch (Exception ex)
+                var feedCleaner = scope.ServiceProvider.GetRequiredService<FeedCleaner>();
+                await feedCleaner.CleanFeedAsync(feed);
+                await Task.CompletedTask;
+            });
+
+            _logger.Log(
+                feedsCleaned != packageFeeds.Count ? LogLevel.Warning : LogLevel.Information,
+                "Successfully processed {count}/{totalCount} package feeds for {account}",
+                feedsCleaned,
+                packageFeeds.Count,
+                azdoAccount);
+
+            _logger.LogInformation("Processing {feedCount} symbol feeds...",
+                symbolFeeds.Count);
+
+            // We refresh the feed information so that symbol feeds can be cleaned based on up-to-date package feeds
+            (packageFeeds, symbolFeeds) = await FetchFeeds(azdoAccount);
```

Comment:
do we really need to do this? doesn't this just give a list of feeds, which will be the same since we didn't delete any package feeds? We get the contents of the feeds here https://github.com/dotnet/arcade-services/blob/5fed2f41ae94a52044756f07073b102ac4e9b90f/src/ProductConstructionService/ProductConstructionService.FeedCleaner/FeedCleaner.cs#L38

Reply:
Ahh I see, we get number of packages here, so this needs to update, my bad